app/views/criteria/_boot.js.erb

```diff
@@ -28,4 +28,41 @@
       });
     }
   });
+  function add_level(id) {
+    var rubric_form = $('#' + id);
+    var last_level = $('#' + id + " .level").first();
+    var new_level = last_level.clone();
+
+    // use regular expressions to update level numbers
+    var regex = new RegExp(/\d+/);
+    var level_num = $('#' + id + " .level").length;
+    // replace the new level names with the new level number
+    var new_name = $(new_level)[0].childNodes[1];
+    $(new_name).text(new_name.textContent.replace(regex, level_num));
+    var display_name = new_level.find('label')[0];
+    $(display_name).text(display_name.textContent.replace(regex, level_num));
+
+    // clear the inputs and text areas and create unique ids and names
+    new_level.find('input').each(function () {
+      if ($(this).attr('id')) {
+        $(this).attr('id', $(this).attr('id').replace(regex, level_num));
+      }
+      if ($(this).attr('name')) {
+        $(this).attr('name',
+                          $(this).attr('name').replace(regex, level_num));
```

Comment:
The indentation here is off. But you can actually fit it in one line, so I'd recommend just moving this up.